### ISSUE TITLE
Multiple proof upload page

### ISSUE THREAD

**Issue by raphodn** [openfoodfacts/open-prices-frontend#983]
### Story

Following #639 where we added a dedicated "single proof upload" page.
Some users want to upload multiple proofs at the same time (proofs with the same type, location, date & currency = Price tags)

### How

A dedicated page.
Loop on all selected images and upload them one by one.

ProofImageInputRow < ProofInputRow < AddPriceMultiple / AddPriceSingle / ProofAddSingle / (ProofAddMultiple)

### Todo

- [x] new page that allows uploading multiple proofs (proof upload form)
- [x] rename card title if multiple mode : https://github.com/openfoodfacts/open-prices-frontend/commit/d87c8cbfe49299ea15ba3210fc9b35e23f90cb31
- [x] restrict the proofs to PRICE_TAG
- [x] show warning message that all the images should share the same location & date
- [ ] check that all the images share the same date
- [x] wait for the upload to be done before activating the bottom CTA (#1189)
- [x] show page in the sidebar

**Comment by raphodn** [comment]
@serpico I just deployed #1043 in staging, if you want to test it out !
https://prices.openfoodfacts.net/proofs/add/multiple

One big caveat I think, is that it doesn't check each image exif if they are indeed from the same date.

**Comment by serpico** [comment]
I gave it a first try here are a few thoughts/feedback : 

- I think we need a note or something to tell the user the "image level" is where it's at ( see screenshot ), I was looking for something at the bottom of the page for how for how to trigger the 2nd, n upload.

![Image](https://github.com/user-attachments/assets/8f560f9b-1a16-4348-86fd-ccd01f5c221f)

- From what I understood I selected 2 pictures ( ctrl + clic ) after clicking on "Select from device gallery" it went smoothly at first but after more careful observation only one appears in the proof ( appear to be effectively uploaded ) , cue only one image appears on the "zoomed preview" ( step 2 - after selecting the pics and pressing Done ). 
 
Q : In case it works and X images are selected and validated in the file manager window should we have an equal number of X of "zoomed preview" images appearing here, like "stacked" ?

- On a second try I got a few error pop pups ( see below )

![Image](https://github.com/user-attachments/assets/b5fb810c-d8e2-4097-ad21-085c62ca3850)

![Image](https://github.com/user-attachments/assets/f48e2160-e5b2-4556-81d5-db286f401da3)

I'll wait for your answer/advice before going any further

**Comment by raphodn** [comment]
ok @serpico i did a quick fix ! the code isn't pretty, but it should work now 🙏

**Comment by serpico** [comment]
There must be something that eludes me here somehow.

I ( think ) do all the things required ( type, image, location, etc...) then the only button I have is Upload, I click it.

I type up the EAN, the price 

![Image](https://github.com/user-attachments/assets/c6941c58-a663-4d35-84e4-0eeebec713ee)

And I press Upload


On the following page I have the same picture ( of course, I wasn't able to upload any other ), but the price field is reset/virginized although the price I just added appears  at the top of the right part of the screen as if I could enter another product and another price for the same picture ?

![Image](https://github.com/user-attachments/assets/d378ab6e-cbbd-4db7-9478-13acd74f6bbb)

This feels more like "Upload multiples prices from the same picture" rather than "Multiple proof upload" don't it ?

TLDR: where is the button to upload another proof image, X other proofs image ?

**Comment by raphodn** [comment]
Wait you're not on the right page 😅 

I'm talking about a new-dedicated-experiment page. Experiments > Add multiple proofs.
Actually it shouldn't be possible to select multiple images elsewhere... 🤔

**Comment by serpico** [comment]
> Wait you're not on the right page 😅
> 
> I'm talking about a new-dedicated-experiment page. Experiments > Add multiple proofs.
> 
> Actually it shouldn't be possible to select multiple images elsewhere... 🤔

I think I was on the right page

**Comment by serpico** [comment]
This morning I tried to upload 22 proof images in one go.

-  As far as I can tell there might be some delay in the images being visible in dashboard > proofs. I had time to input 10 prices and only those 10 proofs were visible, I thought maybe the others weren't uploaded, but I reloaded the page ( either F5 or CTRL + R , I don't remember ) and another batch of 10 were visible - at least we need to warn the user to avoid confusion.

-  Is there a way to clearly define/ensure the order in which the images will be uploaded ( and appear in dashoard > proof ) ? ( based on the order they appear in the previews ( step 1 and 2 ) ( see screenshots ) - or even better, based on the order the user select them in the file manager ( i.e in chronological or reverse chronological order for example )  For some user it could be significant ( when you have a list of EAN you want to process from top to bottom - not secara amburadul ).
 
In fact this probably need further investigation, the order seems not even to be the same between the 1st and the 2nd preview...compare the 2 screenshot, it's the same batch, step 1 and 2, in step 1 the mascarpone ( plastic pot green color ) was at the top, in step 2 it's in 3rd, finally in step 3 (  dashoard > proof ) it's 4th in the order ( fondue, arachides coques x2 which pictures are inverted/upside down and finally, relegated off the podium our old friend mascarpone ) - notice also the position of the inverted price tag, changing at each preview step. 

**Preview step 1** 
![Image](https://github.com/user-attachments/assets/dc6c8df1-97ed-4e32-8653-85732e94db2f)

**Preview step 2** I took 3 screenshots, from the top of the page down so we can see/follow the order of the preview.
![Image](https://github.com/user-attachments/assets/40ba93b3-1994-47aa-ac78-28d5e2877496)

![Image](https://github.com/user-attachments/assets/48e8128c-2887-40e6-b3c0-20d25eaf888e)

![Image](https://github.com/user-attachments/assets/43e8ae85-ffdb-4957-8045-42c3a757572d)

**Dashboard > Proofs** the red line indicate the beginning of this batch, pic on the left of the red line, bottom row is the 1st of this batch, all the pictures on the top row follow from right to left.
![Image](https://github.com/user-attachments/assets/6b84a3ac-fd74-4da2-adb8-b094c96dc6ff)


Recap of the respective order by step :

Step 1 : mascarpone, fondue, raclette, raclette, cacahuètes

Step 2 :  fondue, cacahuètes, mascarpone

Dashboard > proofs : fondue, cacahuètes, cacahuètes, mascarpone

**Comment by raphodn** [comment]
Todo 
- show an info (or warning ?) message to remind users to send images from the same location + date
- raise an error if images have different dates
- investigate the ordering post-upload

**Comment by thibaultmol** [comment]
Is it possible that all the images get loaded into ram in one go or something? 
I'm not sure what's going on but I can say that on my phone firefox grinds to a halt and eventually the tab crashes if i try to add my 17 images which combined are 177MB (they're very high res)

**Comment by raphodn** [comment]
The code loops on each image, compresses them into .webp, and sends them to the server. 

My photos are usually 10MB big, it's a bit slow but I didn't actually measure the split between the compression & the actual uploading to the server.

The photos displayed before upload are the raw/full-size so the browser might have a hard time at that moment ?

**Comment by thibaultmol** [comment]
I don't know what's going on, but the memory spiking is def happening because of the upload. (ignore my extremely high cpu usage, my pc is doing other things in the backgrounds atm. but it's unrelated, as i could repro this as well when my cpu usage was normal)

https://github.com/user-attachments/assets/827d321c-d014-45be-980f-d2d27b34bd71

here are my images: https://drive.proton.me/urls/PT8X329AWW#CBGZzMfgoh4k

**Comment by raphodn** [comment]
Linked to #1047

**Comment by raphodn** [comment]
From the discussions we're having with the rest of the team, and the speed we're going with integrating assisted contribution, we are rather focused at having a seperation between proof upload & price upload.
- these proofs wil arrive here for instance: https://github.com/openfoodfacts/open-prices-frontend/issues/1115
- soon these proofs will have their price & barcode auto-extracted, and users will be presented with individual price tags instead
- and we'll just need users to validate/amend on a per-price tag level

**Comment by serpico** [comment]
Thanks for the clarification

**Comment by raphodn** [comment]
Recent changes on this page: 
- we've decided to restrict to only Price tags to skip a step (never gonna upload multiple receipts for the same shop on the same day ^^) 
- show an info message to remind the location/date unicity (we don't do the date check yet...)
- these proofs are now the only source for the #1137

**Comment by raphodn** [comment]
How it currently looks like

|Blank form|Filled form|Successfully upload|
|---|---|---|
|![Image](https://github.com/user-attachments/assets/918148a6-c60f-4d2a-b84a-f619bf9c4dc2)|||

**Comment by raphodn** [comment]
Closing as this is pretty much done
